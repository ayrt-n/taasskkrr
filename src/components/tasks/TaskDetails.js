import React from 'react';
import closeIcon from '../../assets/icons/close.svg';
import Priority from './Priority';
import Button from '../form/Button';

function TaskDetails({ task, closeModal, editTask }) {
  return (
    <>
      <div className="Modal-header-container">
        <h2>{task.title}</h2>
        <button className="Close-modal-button" onClick={closeModal}>
          <img src={closeIcon} alt="" />
        </button>
      </div>
      <div className="Modal-content-container">
        <div className="field">
          <p className="label">Description</p>
          <p>{task.description}</p>
        </div>
        <div className="field">
          <p className="label">Priority</p>
          <Priority priorityEnum={task.priority} />
        </div>
        <div className="field">
          <p className="label">Due Date</p>
          <p>{task.due_date || "N/A"}</p>
        </div>
        <Button label="Edit Task" buttonStyles="is-primary" onClick={editTask}/>
      </div>
    </>
  );
}

export default TaskDetails;
