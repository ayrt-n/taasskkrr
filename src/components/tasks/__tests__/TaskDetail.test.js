import React from 'react';
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event';
import TaskDetails from '../TaskDetails';

// Set up children to mock
jest.mock('../Priority', () => ({ priorityEnum }) => (
  <div>{priorityEnum}</div>
));

jest.mock('../../form/Button', () => ({ onClick }) => (
  <button onClick={onClick}>Edit Task</button>
));

// Set up props
const closeModalMock = jest.fn();
const editTaskMock = jest.fn();
const taskWithDueDate = {
  title: 'Test Task',
  description: 'Test task description',
  priority: 0,
  due_date: 'Some time in the future',
}
const taskNoDueDate = {
  title: 'Test Task',
  description: 'Test task description',
  priority: 0,
  due_date: null,
}

describe('TaskDetail component', () => {
  it('calls closeModal when close modal button clicked', () => {
    render(
      <TaskDetails
        task={taskWithDueDate}
        closeModal={closeModalMock}
        editTask={editTaskMock}
      />
    );

    const closeButton = screen.getByRole('button', { name: /close modal/i });
    userEvent.click(closeButton);

    expect(closeModalMock).toHaveBeenCalled();
  });

  it('calls editTask when edit task button is clicked', () => {
    render(
      <TaskDetails
        task={taskWithDueDate}
        closeModal={closeModalMock}
        editTask={editTaskMock}
      />
    );

    const editButton = screen.getByRole('button', { name: /edit task/i });
    userEvent.click(editButton);

    expect(editTaskMock).toHaveBeenCalled();
  });

  describe('when provided with a due date', () => {
    it('renders correctly', () => {
      const { container } = render(
        <TaskDetails
          task={taskWithDueDate}
          closeModal={closeModalMock}
          editTask={editTaskMock}
        />
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('when not provided with a due date', () => {
    it('renders correctly', () => {
      const { container } = render(
        <TaskDetails
          task={taskNoDueDate}
          closeModal={closeModalMock}
          editTask={editTaskMock}
        />
      );

      expect(container).toMatchSnapshot();
    });
  });
});