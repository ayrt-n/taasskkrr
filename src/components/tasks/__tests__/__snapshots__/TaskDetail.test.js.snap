// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetail component when not provided with a due date renders correctly 1`] = `
<div>
  <div
    class="Modal-header-container"
  >
    <h2>
      Test Task
    </h2>
    <button
      aria-label="close modal"
      class="Close-modal-button"
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </button>
  </div>
  <div
    class="Modal-content-container"
  >
    <div
      class="field"
    >
      <p
        class="label"
      >
        Description
      </p>
      <p>
        Test task description
      </p>
    </div>
    <div
      class="field"
    >
      <p
        class="label"
      >
        Priority
      </p>
      <div>
        0
      </div>
    </div>
    <div
      class="field"
    >
      <p
        class="label"
      >
        Due Date
      </p>
      <p>
        N/A
      </p>
    </div>
    <button>
      Edit Task
    </button>
  </div>
</div>
`;

exports[`TaskDetail component when provided with a due date renders correctly 1`] = `
<div>
  <div
    class="Modal-header-container"
  >
    <h2>
      Test Task
    </h2>
    <button
      aria-label="close modal"
      class="Close-modal-button"
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </button>
  </div>
  <div
    class="Modal-content-container"
  >
    <div
      class="field"
    >
      <p
        class="label"
      >
        Description
      </p>
      <p>
        Test task description
      </p>
    </div>
    <div
      class="field"
    >
      <p
        class="label"
      >
        Priority
      </p>
      <div>
        0
      </div>
    </div>
    <div
      class="field"
    >
      <p
        class="label"
      >
        Due Date
      </p>
      <p>
        Some time in the future
      </p>
    </div>
    <button>
      Edit Task
    </button>
  </div>
</div>
`;
